src/plugins/nestedHeaders/stateManager/sourceSettings.js

```diff
@@ -0,0 +1,197 @@
+import { extend, isObject } from '../../../helpers/object';
+import { arrayEach } from '../../../helpers/array';
+import { settingsNormalizer } from './settingsNormalizer';
+import { HEADER_CONFIGURABLE_PROPS } from './constants';
+
+/**
+ * The class manages and normalizes settings passed by the developer
+ * into the nested headers plugin. The SourceSettings class is a
+ * source of truth for tree builder (HeaderTree) module.
+ *
+ * @class {SourceSettings}
+ */
+export default class SourceSettings {
+  /**
+   * The normalized source data (normalized user-defined settings for nested headers).
+   *
+   * @private
+   * @type {Array[]}
+   */
+  #data = [];
+  /**
+   * The total length of the nested header layers.
+   *
+   * @private
+   * @type {number}
+   */
+  #dataLength = 0;
```

Comment:
Does it need an extra property?

Reply:
Yes, the property is used several times in the same file. I treat this technique as a shortcut which by compressing the code it increases its readability.

Reply:
For me, `data.length` would be also clear and we won't have to set this variable separately :)